src/amqp-ts.ts

```diff
@@ -343,26 +343,33 @@ export class Message {
     return content;
   }
 
-  sendTo(destination: Exchange | Queue, routingKey: string = ""): void {
+  sendTo(destination: Exchange | Queue, routingKey: string = ""): Promise<any> {
     // inline function to send the message
     var sendMessage = () => {
-      try {
-        destination._channel.publish(exchange, routingKey, this.content, this.properties);
-      } catch (err) {
-        log.log("debug", "Publish error: " + err.message, { module: "amqp-ts" });
-        var destinationName = destination._name;
-        var connection = destination._connection;
-        log.log("debug", "Try to rebuild connection, before Call.", { module: "amqp-ts" });
-        connection._rebuildAll(err).then(() => {
-          log.log("debug", "Retransmitting message.", { module: "amqp-ts" });
-          if (destination instanceof Queue) {
-            connection._queues[destinationName].publish(this.content, this.properties);
+      return new Promise((resolve, reject) => {
+        destination._channel.publish(exchange, routingKey, this.content, this.properties, (err, data) => {
+          if (!err) {
+            resolve(data);
```

Comment:
The example [here](https://github.com/squaremo/amqp.node/blob/31bb2a9c40219ce2fc8c4fede15a84a9f54b36a1/examples/waitForConfirms.js#L17) suggests you need to await `_channel.waitForConfirms`

Reply:
I don't know why they combine the callback with `waitForConfirms` but I assure you it's not needed :P